### PR TITLE
Migrate "Understanding constraints" to null safety

### PR DESCRIPTION
**Pull request by miquelbeltran**
Here I migrated the constraints.md to null safety following this:

- Moved the source code to a project in the `null_safety_examples`
- Used code excerpts to provide all the code in the article
- Also enabled null safety to the embedded DartPad

the only two sections of code that has not been migrated to the code-excerpt extractor are the ones that reference the framework, at the end in the examples of the `BoxConstraint`.

@sfshaza2 @RedBrogdon 

Also pinging @marcglasberg, as the article references your repo: https://github.com/marcglasberg/flutter_layout_article, just in case you want to upgrade the code with the null safety changes!

### CODE REVIEW

**Review by sfshaza2 — COMMENTED** [review]
@domesticmouse, can you review? It looks good to me.

@miquelbeltran, just an FYI. All of Google has tomorrow (Friday) off.

File: src/docs/development/ui/layout/constraints.md [hunk 1]
```diff
@@ -227,7 +229,8 @@ class _FlutterLayoutArticleState extends State<FlutterLayoutArticle> {
                     Expanded(
                         child: ConstrainedBox(
                             constraints: BoxConstraints.tightFor(
-                                width: double.infinity, height: double.infinity),
+                                width: double.infinity,
+                                height: double.infinity),
```
Comment (domesticmouse) [inline src/docs/development/ui/layout/constraints.md:233]: nit: add trailing `,` for nicer formatting?

File: src/docs/development/ui/layout/constraints.md [hunk 2]
```diff
@@ -417,7 +424,8 @@ class Example5 extends Example {
   @override
   Widget build(BuildContext context) {
     return Center(
-      child: Container(width: double.infinity, height: double.infinity, color: red),
+      child: Container(
+          width: double.infinity, height: double.infinity, color: red),
```
Comment (domesticmouse) [inline src/docs/development/ui/layout/constraints.md:428]: I'd also add a trailing `,` here as well

File: src/docs/development/ui/layout/constraints.md [hunk 3]
```diff
@@ -521,7 +528,8 @@ class Example9 extends Example {
   @override
   Widget build(BuildContext context) {
     return ConstrainedBox(
-      constraints: BoxConstraints(minWidth: 70, minHeight: 70, maxWidth: 150, maxHeight: 150),
+      constraints: BoxConstraints(
+          minWidth: 70, minHeight: 70, maxWidth: 150, maxHeight: 150),
```
Comment (domesticmouse) [inline src/docs/development/ui/layout/constraints.md:532]: ditto on trailing `'`

File: src/docs/development/ui/layout/constraints.md [hunk 4]
```diff
@@ -547,7 +555,8 @@ class Example10 extends Example {
   Widget build(BuildContext context) {
     return Center(
       child: ConstrainedBox(
-        constraints: BoxConstraints(minWidth: 70, minHeight: 70, maxWidth: 150, maxHeight: 150),
+        constraints: BoxConstraints(
+            minWidth: 70, minHeight: 70, maxWidth: 150, maxHeight: 150),
```
Comment (domesticmouse) [inline src/docs/development/ui/layout/constraints.md:559]: ditto

File: src/docs/development/ui/layout/constraints.md [hunk 5]
```diff
@@ -573,7 +582,8 @@ class Example11 extends Example {
...
```
Comment (domesticmouse) [inline src/docs/development/ui/layout/constraints.md:586]: ditto

File: src/docs/development/ui/layout/constraints.md [hunk 6]
```diff
@@ -1114,14 +1171,11 @@ available space.
...
+<?code-excerpt "lib/main.dart (Example5)" replace="/(return |;)//g"?>
 ```dart
 Center(
-   child: Container(
-      color: Colors.red,
-      width: double.infinity,
-      height: double.infinity,
-   )
+  child: Container(
+      width: double.infinity, height: double.infinity, color: red),
```
Comment (domesticmouse) [inline src/docs/development/ui/layout/constraints.md:1178]: any particular reason to drop trailing `,`?

File: src/docs/development/ui/layout/constraints.md [hunk 7]
```diff
@@ -1324,10 +1368,10 @@ since that’s between 70 and 150.
 
 {% asset ui/layout/layout-13.png class="mw-100" alt="Example 13 layout" %}
 
-<!-- skip -->
+<?code-excerpt "lib/main.dart (Example13)" replace="/(return |;)//g"?>
 ```dart
 UnconstrainedBox(
-   child: Container(color: Colors.red, width: 20, height: 50),
+  child: Container(color: red, width: 20, height: 50),
```
Comment (domesticmouse) [inline src/docs/development/ui/layout/constraints.md:1374]: I'm curious where the `red` here comes from, if not from `Colors`?

File: src/docs/development/ui/layout/constraints.md [hunk 8]
```diff
@@ -1340,10 +1384,10 @@ lets its child `Container` be any size it wants.
 
 {% asset ui/layout/layout-14.png class="mw-100" alt="Example 14 layout" %}
 
-<!-- skip -->
+<?code-excerpt "lib/main.dart (Example14)" replace="/(return |;)//g"?>
 ```dart
 UnconstrainedBox(
-   child: Container(color: Colors.red, width: 4000, height: 50),
+  child: Container(color: red, width: 4000, height: 50),
```
Comment (domesticmouse) [inline src/docs/development/ui/layout/constraints.md:1390]: ditto

File: src/docs/development/ui/layout/constraints.md [hunk 9]
```diff
@@ -1360,15 +1404,15 @@ the much dreaded "overflow warning".
...
-   minHeight: 0.0,
-   maxWidth: double.infinity,
-   maxHeight: double.infinity,
-   child: Container(color: Colors.red, width: 4000, height: 50),
-);
+  minWidth: 0.0,
+  minHeight: 0.0,
+  maxWidth: double.infinity,
+  maxHeight: double.infinity,
+  child: Container(color: red, width: 4000, height: 50),
```
Comment (domesticmouse) [inline src/docs/development/ui/layout/constraints.md:1414]: likewise

File: src/docs/development/ui/layout/constraints.md [hunk 10]
```diff
@@ -1544,13 +1587,13 @@ and you'll see an error in the console.
...
-<!-- skip -->
+<?code-excerpt "lib/main.dart (Example23)" replace="/(return |;)//g"?>
 ```dart
 Row(
-   children:[
-      Container(color: Colors.red, child: Text('Hello!')),
-      Container(color: Colors.green, child: Text('Goodbye!')),
-   ]
+  children: [
+    Container(color: red, child: Text('Hello!', style: big)),
```
Comment (domesticmouse) [inline src/docs/development/ui/layout/constraints.md:1594]: Where is the `big` style coming from?

File: src/docs/development/ui/layout/constraints.md [hunk 11]
```diff
@@ -1567,13 +1610,16 @@ and any extra space remains empty.
...
+<?code-excerpt "lib/main.dart (Example24)" replace="/(return |;)//g"?>
 ```dart
 Row(
-   children:[
-      Container(color: Colors.red, child: Text('This is a very long text that won’t fit the line.')),
-      Container(color: Colors.green, child: Text('Goodbye!')),
-   ]
+  children: [
+    Container(
+        color: red,
```
Comment (domesticmouse) [inline src/docs/development/ui/layout/constraints.md:1618]: ditto

**Review by domesticmouse — APPROVED** [review]
Bunch of nits

**Comment by sfshaza2** [comment]
Thanks, @miquelbeltran and @domesticmouse!

**Pull request merged**